src/core/transpile.jl

```diff
@@ -2579,3 +2579,143 @@ function are_gates_at_excluded_positions(
 
     return (false, "")
 end
+
+struct RejectGatesOnExcludedConnectionsTranspiler <: Transpiler
+    connectivity::AbstractConnectivity
+end
+
+"""
+    transpile(
+        transpiler::RejectGatesOnExcludedConnectionsTranspiler,
+        circuit::QuantumCircuit
+    )::QuantumCircuit
+
+Throws a `DomainError` if an `Instruction` in the `circuit` operates on an excluded
+connection. The excluded connections are specified with the parameter `excluded_connections`
+in certain `AbstractConnectivity` objects. The `circuit` remains unchanged if no error is
```

Comment:
The `circuit` remains unchanged.

Reply:
(you seem to imply otherwise)

Reply:
I will replace the last sentence with "The function returns the same `circuit` if no error is thrown."